File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        int add1 = 0;
        int add2 = 0; 
        int mul1 = 1; 
        int mul2 = 1; 
        
        for (int i = 0; i < num_list.length; i++) {
            if (num_list[i]%2 != 0) {
                add1++;
            } else {
                add2++; 
            }
        }
        
        for (int i = 0; i < add1; i++) {
            mul1 *= 10;
        }
        for (int i = 0; i < add2; i++) {
            mul2 *= 10;
        }
        mul1 /= 10;
        mul2 /= 10;
        
        add1 = 0;
        add2 = 0; 
        
        for (int i = 0; i < num_list.length; i++) {
            if (num_list[i]%2 != 0) {
                add1 += num_list[i] * mul1;
                mul1 /= 10;
            } else {
                add2 += num_list[i] * mul2;
                mul2 /= 10; 
            }
        }
        return add1+add2;
    }
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// num_list_len은 배열 num_list의 길이입니다.
int solution(int num_list[], size_t num_list_len) {
    int h = 0; 
    int j = 0; 
    int h_mul = 1; 
    int j_mul = 1; 
    for (int i = num_list_len-1; i >= 0; i--) {
        if (num_list[i]%2 == 0) {
            j += num_list[i]*j_mul;
            j_mul *= 10;
        } else {
            h += num_list[i]*h_mul;
            h_mul *= 10; 
        }
    }
    return j+h;
}